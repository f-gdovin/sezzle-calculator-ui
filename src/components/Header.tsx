import React, { FunctionComponent } from 'react';
import { ReactComponent as ReactLogo } from '../logo.svg';
import styles from '../styles.module.css';

interface HeaderProps {
    onClick: Function;
}

const areEqual = (prevProps: any, nextProps: any) => true;

const Header: FunctionComponent<HeaderProps> = React.memo((props: HeaderProps) => {
        return <ReactLogo
            className={styles['pointy']}
            height='5em'
            preserveAspectRatio="xMinYMin"
            onClick={() => props.onClick()}/>
    }, areEqual);

export default Header;