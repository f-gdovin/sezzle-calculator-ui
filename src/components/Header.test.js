import React from 'react';
import renderer from 'react-test-renderer';
import Header from './Header';

test('should render Header component that looks as expected', () => {
    const tree = renderer
        .create(<Header onClick={() => console.log('Clicked on logo')}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});