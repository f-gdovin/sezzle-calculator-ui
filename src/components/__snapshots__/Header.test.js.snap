// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header component that looks as expected 1`] = `
<svg
  className="pointy"
  height="5em"
  onClick={[Function]}
  preserveAspectRatio="xMinYMin"
>
  logo.svg
</svg>
`;
